collector/cs.go

```diff
@@ -79,5 +93,21 @@ func (c *CSCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, er
 		float64(dst[0].TotalPhysicalMemory),
 	)
 
+	var fqdn string
+	if len(dst[0].Domain) > 0 {
+		fqdn = dst[0].DNSHostname + "." + dst[0].Domain
+	} else {
+		fqdn = dst[0].DNSHostname
+	}
```

Comment:
When is the domain empty? From some quick testing, `Domain` is the workgroup name on non-domain joined machines, are there more cases here I'm missing?

Reply:
I have only tried it on domain joined machines, I just assumed that `Domain` would be empty as there is another `Workgroup` field in that class.

I don't know if there are cases in which this could be empty. You are right that this is the Workgroup name on non-domain joined machines. 

Also I don't know if it would be expected behavior that the fqdn on such machines would then be `Hostname.WORKGROUP` in the metric. Maybe I should change this check to be more like this
```go
if dst[0].Domain != dst[0].Workgroup {
        fqdn = dst[0].DNSHostname + "." + dst[0].Domain
} else {
	fqdn = dst[0].DNSHostname
}
```
What do you think?

Reply:
That looks reasonable! Or we could skip the fqdn altogether? It is quite simple to add via relabeling if it is needed, and I'm not sure how many users would have multiple AD domains scraped by the same Prometheus instance?

Reply:
I adapted the check.
I wanted to include the fqdn for dashboarding reasons. It is much easier to make a Grafana Dashboard with dropdowns for the hosts (e.g. https://grafana.com/grafana/dashboards/1860) with the availability of such a metric. So i would like to keep it even if it might seem redundant.